// ngx-countdown/esm2015/src/countdown.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { CountdownComponent } from './countdown.component';
import { Timer } from './countdown.timer';
import * as ɵngcc0 from '@angular/core';
export class CountdownModule {
}
CountdownModule.ngModuleDef = ɵngcc0.ɵɵdefineNgModule({ type: CountdownModule });
/*@__PURE__*/ ɵngcc0.ɵɵsetNgModuleScope(CountdownModule, { declarations: function () { return [CountdownComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CountdownComponent]; } });
/*@__PURE__*/ ɵngcc0.ɵsetClassMetadata(CountdownModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                providers: [Timer],
                declarations: [CountdownComponent],
                exports: [CountdownComponent]
            }]
    }], null, null);
CountdownModule.ngInjectorDef = ɵngcc0.ɵɵdefineInjector({ factory: function CountdownModule_Factory(t) { return new (t || CountdownModule)(); }, providers: [Timer], imports: [[CommonModule]] });
