// angular2-busy/build/busy-backdrop.component.js
/**
 * @file Component: BusyBackdrop
 * @author yumao<yuzhang.lille@gmail.com>
 */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var animations_1 = require("@angular/animations");
var promise_tracker_service_1 = require("./promise-tracker.service");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@angular/common');

const _c0 = ["class", "ng-busy-backdrop", 4, "ngIf"];
const _c1 = [1, "ng-busy-backdrop", 3];
function BusyBackdropComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", _c1);
} if (rf & 2) {
    ɵngcc0.ɵɵproperty("@fadeInOut", undefined);
} }
var inactiveStyle = animations_1.style({
    opacity: 0,
});
var timing = '.3s ease';
var BusyBackdropComponent = (function () {
    function BusyBackdropComponent(tracker) {
        this.tracker = tracker;
    }
    BusyBackdropComponent.prototype.isActive = function () {
        return this.tracker.isActive();
    };
BusyBackdropComponent.ngComponentDef = ɵngcc0.ɵɵdefineComponent({ type: BusyBackdropComponent, selectors: [["ng-busy-backdrop"]], factory: function BusyBackdropComponent_Factory(t) { return new (t || BusyBackdropComponent)(ɵngcc0.ɵɵdirectiveInject(promise_tracker_service_1.PromiseTrackerService)); }, consts: 1, vars: 1, template: function BusyBackdropComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, BusyBackdropComponent_div_0_Template, 1, 1, "div", _c0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.isActive());
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2, data: { animation: [
            animations_1.trigger('fadeInOut', [
                animations_1.transition('void => *', [
                    inactiveStyle,
                    animations_1.animate(timing)
                ]),
                animations_1.transition('* => void', [
                    animations_1.animate(timing, inactiveStyle)
                ])
            ])
        ] } });
/*@__PURE__*/ ɵngcc0.ɵsetClassMetadata(BusyBackdropComponent, [{
        type: core_1.Component,
        args: [{
                selector: 'ng-busy-backdrop',
                template: "\n        <div class=\"ng-busy-backdrop\"\n             @fadeInOut\n             *ngIf=\"isActive()\">\n        </div>\n    ",
                animations: [
                    animations_1.trigger('fadeInOut', [
                        animations_1.transition('void => *', [
                            inactiveStyle,
                            animations_1.animate(timing)
                        ]),
                        animations_1.transition('* => void', [
                            animations_1.animate(timing, inactiveStyle)
                        ])
                    ])
                ]
            }]
    }], function () { return [{ type: promise_tracker_service_1.PromiseTrackerService }]; }, { tracker: [], isActive: [] });
    return BusyBackdropComponent;
}());
/** @nocollapse */
BusyBackdropComponent.ctorParameters = function () { return [
    { type: promise_tracker_service_1.PromiseTrackerService, },
]; };
exports.BusyBackdropComponent = BusyBackdropComponent;
