// angular2-busy/build/busy.component.js
/**
 * @file Component: Busy
 * @author yumao<yuzhang.lille@gmail.com>
 */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var animations_1 = require("@angular/animations");
var promise_tracker_service_1 = require("./promise-tracker.service");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@angular/common');

const _c0 = [3, "class", 4, "ngIf"];
const _c1 = [4, "ngComponentOutlet", "ngComponentOutletNgModuleFactory"];
function BusyComponent_div_0_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function BusyComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵstyling();
    ɵngcc0.ɵɵtemplate(1, BusyComponent_div_0_ng_container_1_Template, 1, 0, "ng-container", _c1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1285 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(ctx_r1285.wrapperClass);
    ɵngcc0.ɵɵstylingApply();
    ɵngcc0.ɵɵproperty("@flyInOut", undefined);
    ɵngcc0.ɵɵselect(1);
    ɵngcc0.ɵɵproperty("ngComponentOutlet", ctx_r1285.TemplateComponent)("ngComponentOutletNgModuleFactory", ctx_r1285.nmf);
} }
var inactiveStyle = animations_1.style({
    opacity: 0,
    transform: 'translateY(-40px)'
});
var timing = '.3s ease';
;
var BusyComponent = (function () {
    function BusyComponent(tracker, compiler) {
        this.tracker = tracker;
        this.compiler = compiler;
    }
    BusyComponent.prototype.ngDoCheck = function () {
        if (this.message === this.lastMessage) {
            return;
        }
        this.lastMessage = this.message;
        this.clearDynamicTemplateCache();
        this.createDynamicTemplate();
    };
    BusyComponent.prototype.ngOnDestroy = function () {
        this.clearDynamicTemplateCache();
    };
    BusyComponent.prototype.createDynamicTemplate = function () {
        var _a = this, template = _a.template, message = _a.message;
        var TemplateComponent = (function () {
            function TemplateComponent() {
                this.message = message;
            }
            return TemplateComponent;
        }());
        TemplateComponent.decorators = [
            { type: core_1.Component, args: [{ template: template },] },
        ];
        /** @nocollapse */
        TemplateComponent.ctorParameters = function () { return []; };
        var TemplateModule = (function () {
            function TemplateModule() {
            }
            return TemplateModule;
        }());
        TemplateModule.decorators = [
            { type: core_1.NgModule, args: [{
                        declarations: [TemplateComponent],
                        entryComponents: [TemplateComponent]
                    },] },
        ];
        /** @nocollapse */
        TemplateModule.ctorParameters = function () { return []; };
        this.TemplateComponent = TemplateComponent;
        this.nmf = this.compiler.compileModuleSync(TemplateModule);
    };
    BusyComponent.prototype.clearDynamicTemplateCache = function () {
        if (!this.nmf) {
            return;
        }
        this.compiler.clearCacheFor(this.nmf.moduleType);
        this.nmf = null;
    };
    BusyComponent.prototype.isActive = function () {
        return this.tracker.isActive();
    };
BusyComponent.ngComponentDef = ɵngcc0.ɵɵdefineComponent({ type: BusyComponent, selectors: [["ng-busy"]], factory: function BusyComponent_Factory(t) { return new (t || BusyComponent)(ɵngcc0.ɵɵdirectiveInject(promise_tracker_service_1.PromiseTrackerService), ɵngcc0.ɵɵdirectiveInject(core_1.Compiler)); }, consts: 1, vars: 1, template: function BusyComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, BusyComponent_div_0_Template, 2, 3, "div", _c0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.isActive());
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgComponentOutlet], encapsulation: 2, data: { animation: [
            animations_1.trigger('flyInOut', [
                animations_1.transition('void => *', [
                    inactiveStyle,
                    animations_1.animate(timing)
                ]),
                animations_1.transition('* => void', [
                    animations_1.animate(timing, inactiveStyle)
                ])
            ])
        ] } });
/*@__PURE__*/ ɵngcc0.ɵsetClassMetadata(BusyComponent, [{
        type: core_1.Component,
        args: [{
                selector: 'ng-busy',
                template: "\n        <div [class]=\"wrapperClass\" *ngIf=\"isActive()\" @flyInOut>\n            <ng-container *ngComponentOutlet=\"TemplateComponent; ngModuleFactory: nmf;\"></ng-container>\n        </div>\n    ",
                animations: [
                    animations_1.trigger('flyInOut', [
                        animations_1.transition('void => *', [
                            inactiveStyle,
                            animations_1.animate(timing)
                        ]),
                        animations_1.transition('* => void', [
                            animations_1.animate(timing, inactiveStyle)
                        ])
                    ])
                ]
            }]
    }], function () { return [{ type: promise_tracker_service_1.PromiseTrackerService }, { type: core_1.Compiler }]; }, { tracker: [], compiler: [], ngDoCheck: [], lastMessage: [], ngOnDestroy: [], createDynamicTemplate: [], TemplateComponent: [], nmf: [], clearDynamicTemplateCache: [], isActive: [] });
    return BusyComponent;
}());
/** @nocollapse */
BusyComponent.ctorParameters = function () { return [
    { type: promise_tracker_service_1.PromiseTrackerService, },
    { type: core_1.Compiler, },
]; };
exports.BusyComponent = BusyComponent;
